javascript/require-and-import-mini-lesson.md

```diff
@@ -0,0 +1,351 @@
+
+So, open the `example-for-imports.js` file and change the first line for this one:
+
+```javascript
+import { firstMessage, secondMessage} from './my-module-for-import.js';
+```
+
+Since we changed the file extension, we need to change it in our import.
+
+Next, we need to create the `package.json` file and put this code in:
+
+```javascript
+{
+  "type": "module"
+}
+```
+
+This simply will lay the grounds for our program to run as a module. To test this, inside of the new created folderwe first need to run the install command like I mentioned above:
+
+```bash
+npm install
+```
+
+NPM will run the installation and you can possibly see warnings or notices, don't mind them, it's fine for our purposes.
+
+When the installation is done, you can now run:
+
+```bash
+node example-for-import.js
+```
+
+And you should see:
+
+```bash
+This is a the first message!
+This is a the second message!
+```
+
+That's it! You now know how to use `import` with Node!
+
+Please note that the key thing with this is to always stay tuned to changes in the ECMA specifications, as they change often and the codebase progresses to bring in usually more depth and better performance. In my opinion, a good developer is a developer who's always keen to learn and is on the lookout for these changes!
+
+Now, pat yourself on the back, as you've gone through somewhat tough material. Go back, revise, try different things, and when you're ready, tackle on some of the challenges below!
+
+### Challenges
+
+- How could you change the line `const obj = require("./module-with-functions");` and change it to become more dynamic? Can you refactor it by using ES6 syntax? What's the benefit for this?
+
+- Using both the `require` and `import` methods, try exporting something different than the simple Strings we used in the practice, and then import it in your program. It could be a number, it could be an object, it could be a function, etc. How will you call what you have exported? Reflect on the many usage of this and how you could apply this to a real app.
+
+- Try a mix of the imports seen for the `import` method. Which one do you find more useful? Which one do you think is easier to work with?
+
+- Try to create different modules in different paths and call them within the same file.
+
+### Check for Understanding
+
+- Try to recreate the summary chart by yourself, see what information you retained.
+
+- Make a detailed cheatsheet of the syntax for `require` and `import`, clearly differentiating between the different imports for the latest.
+
+
+### Supplemental Materials
+
+- [Official MDN documentation on modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules)
+
+- [Browser Suppor For Modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules#Browser_support)
```

Comment:
This could be a nice challenge that there is module support in browsers to load more than one JS file but it is also seldom used in real apps.

Reply:
removed
[c89273b](https://github.com/Techtonica/curriculum/pull/1414/commits/c89273b844a10c1f67a1d0886a773f18a1c327e8)